src/Peachpie.Library.Network/CURLConstants.cs

```diff
@@ -1067,11 +1086,58 @@ public override void Apply(Context ctx, HttpWebRequest request)
 
         private static void LoadCookieFile(HttpWebRequest request, PhpStream stream)
         {
-            // TODO: load netscape-like or header-style cookies from stream
+            // load netscape-like or header-style cookies from stream
 
-            // request.CookieContainer.Add() // add parsed cookie
+            // TODO: add support for header-style cookies (`Set-Cookie: ...`).
+            //       currently, only curl's netscape-like cookies are supported.
 
             // request.CookieContainer.SetCookies( ... ) // set header style cookies
+
+            using (var reader = new StreamReader(stream.RawStream, Encoding.ASCII))
+            {
+				string line = String.Empty;
+
+				// iterate over the cookie jar lines
+				while ((line = reader.ReadLine()) != null)
+				{
+
+					// gather current cookie information in a new cookie object
+					var cookie = new Cookie();
```

Comment:
the line should be parsed first, verified, and then `Cookie` object created

Reply:
Resolved. This was required for the `HttpOnly` flag, so I used a bool instead.